components/os/os.go

```diff
@@ -1,40 +1,80 @@
 package os
 
-import "github.com/DataDog/test-infra-definitions/components/command"
+import (
+	"fmt"
 
-type Architecture string
-
-const (
-	AMD64Arch = Architecture("x86_64")
-	ARM64Arch = Architecture("arm64")
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components/command"
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
-// The types of OSes that are common
-type Type int
-
-const (
-	UnixType    Type = iota
-	WindowsType Type = iota
-	OtherType   Type = iota
-)
+// Interfaces used by OS components
+type PackageManager interface {
+	Ensure(packageRef string, opts ...pulumi.ResourceOption) (*remote.Command, error)
+}
 
-type AgentVersion struct {
-	Major       string
-	Minor       string // Empty means latest
-	BetaChannel bool
-	PipelineID  string
-	Arch        string
+type ServiceManager interface {
+	// EnsureStarted starts or restarts (may be stop+start depending on implementation) the service if already running
+	EnsureRestarted(serviceName string, triggers pulumi.ArrayInput, opts ...pulumi.ResourceOption) (*remote.Command, error)
 }
 
+// FileManager needs to be added here as well instead of the command package
+
+// OS is the high-level interface for an OS INSIDE Pulumi code
 type OS interface {
-	GetImage(Architecture) (string, error)
-	GetDefaultInstanceType(Architecture) string
-	GetServiceManager() *ServiceManager
-	GetAgentConfigFolder() string
-	GetSSHUser() string
-	GetAgentInstallCmd(AgentVersion) (string, error)
-	GetRunAgentCmd(parameters string) string
-	GetType() Type
-	CreatePackageManager(runner *command.Runner) (command.PackageManager, error)
-	CheckIsAbsPath(string) bool
+	Descriptor() Descriptor
+
+	Runner() *command.Runner
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Should `Runner` be injected, rather than being a member of an instance ?

Reply:
I am not sure I understand this one. `Runner` is already injected in the `os` implementation of the interface, the interface just mandates to have a way to return the `Runner`.

Reply:
If injected, I expect a getter on the injector rather than on the injected. But I can see we might not have access to the injector and needing it on the injected interface.